starknet_py/net/schemas/common.py

```diff
@@ -44,12 +51,67 @@ def _deserialize(
             return value
 
         if not isinstance(value, str) or not value.startswith("0x"):
-            raise ValidationError(f"Invalid value provided for felt: {value}.")
+            raise ValidationError(f"Invalid value provided for Felt: {value}.")
 
         try:
             return int(value, 16)
         except ValueError as error:
-            raise ValidationError("Invalid felt.") from error
+            raise ValidationError("Invalid Felt.") from error
+
+
+class Uint64(fields.Field):
+    """
+    Field that serializes int to RPC u64 (hex encoded string)
+    """
+
+    MAX_VALUE = 2**64
+    REGEX_PATTERN = r"^0x(0|[a-fA-F1-9]{1}[a-fA-F0-9]{0,15})"
+
+    def _serialize(self, value: Any, attr: str, obj: Any, **kwargs):
+        if self._is_int_and_in_range(value):
+            return hex(value)
+
+        if self._is_str_and_valid_pattern(value):
+            return value
+
+        raise ValidationError(
+            f"Invalid value provided for {self.__class__.__name__}: {value}"
+        )
+
+    def _deserialize(
+        self,
+        value: Any,
+        attr: Union[str, None],
+        data: Union[Mapping[str, Any], None],
+        **kwargs,
+    ):
+        if self._is_int_and_in_range(value):
+            return value
+
+        if self._is_str_and_valid_pattern(value):
+            return int(value, 16)
+
+        raise ValidationError(
+            f"Invalid value provided for {self.__class__.__name__}: {value}"
+        )
+
+    def _is_int_and_in_range(self, value: Any) -> bool:
+        return isinstance(value, int) and 0 <= value < self.MAX_VALUE
+
+    def _is_str_and_valid_pattern(self, value: Any) -> bool:
+        return (
+            isinstance(value, str)
+            and re.fullmatch(self.REGEX_PATTERN, value) is not None
+        )
+
+
+class Uint128(Uint64):
```

Comment:
Maybe it would make sense to generalize what's in Uint64 class, and have Uint64 and Uint128 classes directly inherit from this new class? I guess it might be a nitpick thoug

Reply:
Maybe it would even make sense to have a common abstract class for `Felt`, `Uint64` and `Uint128`?

Reply:
If the interface is the same - yes